### DISCUSSION TITLE
Undocumented how to access users dashboard

### DISCUSSION THREAD

**Discussion by HannesJo0139** [authelia/authelia#7129] [Documentation Feedback]
It seems completely undocumented what address I should use to offer my users a link to the users dashboard / settings.

I'm using swag with the default subdirectory config and until Authelia 4.37 directing users to https://mysite/authelia worked like a charm. With the latest updates this suddenly leads to Authelia just presenting an empty page. I have to add a trailing slash to fix. All of this seems also undocumented.

Since I assumed this being a bug you can lookup everything in https://github.com/authelia/authelia/issues/7119. Unfortunately not providing completely unnecessary and useless info seemed to trigger an attitude and immediate closing of the issue. Still I'm happy if I can help anyone not running into this.

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Based on the information provided and the context from the sources, it appears that the issue you're encountering with the need for a trailing slash (`https://mysite/authelia/`) to access the user dashboard/settings in Authelia version 4.38.7 is indeed not explicitly documented. The behavior you've described, where omitting the trailing slash results in an empty page, seems to be related to how URLs are handled in the newer version of Authelia, as indicated in the GitHub issue you referenced ([#7119](https://github.com/authelia/authelia/issues/7119)).

The discussion in the GitHub issue suggests that the behavior regarding the trailing slash is intentional, with URLs `https://example.com/abc` and `https://example.com/abc/` not being treated as equivalent. However, the resolution or workaround for this specific scenario—directing users to the user dashboard/settings—is not clearly documented in the Authelia documentation or in the issue discussion.

Given this situation, the best practice for now would be to ensure that any links you provide to users for accessing the Authelia user dashboard/settings include the trailing slash. This approach should mitigate the issue of users encountering an empty page.

For a more permanent solution or clarification on whether this behavior will be addressed in future updates, it would be advisable to follow up through the official Authelia channels, such as the [GitHub Discussions](https://github.com/authelia/authelia/discussions) for Authelia, where you can raise this as a question or feature request for better handling or documentation of URL paths in Authelia.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment]
We're still happy to help and issue a fix if necessary provided we receive the information that has been provided. It can't be replicated with the information provided, the default redirection with SWAG takes you to the correct path.

**Comment by james-d-elliott** [comment]
Please explain what you mean? The users access their settings the same way as before, just there is more available to them. Accessing their MFA options to register / re register etc.

**Comment by HannesJo0139** [comment]
This whole story has absolutely nothing to do with any auto redirection of the proxy. It is about a hyperlink I am providing on lets say heimdall, where my users can click to go to their Authelia settings. On a subdomain setup it may be auth.my-site.org, on a subfolder setup (swag default) it may be my-site.org/authelia or my-site.org/authelia/ or my-site.org/authelia/settings or my-site.org/authelia/settings/, etc... Who knows?

Even before 4.38 I was not able to find any doc info on where I am supposed to point that hyperlink. The way I did it (my-site.org/authelia) was just a guess. When I reported https://github.com/authelia/authelia/issues/7119 you stated that the trailing slash is not optional anymore so while implicitly confirming the behavior I was wondering where to find that info in the docs.

**Comment by james-d-elliott** [comment]
> This whole story has absolutely nothing to do with any auto redirection of the proxy. It is about a hyperlink I am providing on lets say heimdall, where my users can click to go to their Authelia settings. On a subdomain setup it may be auth.my-site.org, on a subfolder setup (swag default) it may be my-site.org/authelia or my-site.org/authelia/ or my-site.org/authelia/settings or my-site.org/authelia/settings/, etc... Who knows?

Ahhhh! That makes sense. Seemed like it was more to do with the redirection when taking in the full context. It was a simple misunderstanding.

The page is officially the `/settings/` URI I believe at least for now, there's currently an unknown behavior with react-router which causes it to strip the trailing slashes. I can double check the behavior and figure out a location in the docs to put it, though I'm not sure if that's really going to have a lot of benefit as will likely become clear.

> Even before 4.38 I was not able to find any doc info on where I am supposed to point that hyperlink. 

It's in the docs for the server configuration where you actually configure the value. Here are the docs for [v4.37.5](https://github.com/authelia/authelia/blob/v4.37.5/docs/content/en/configuration/miscellaneous/server.md#path) for example. 

> Perhaps now you understand why logs and config were not needed.

I'm afraid that's not factual and I do not agree. The debug logs and full configuration likely would have been incredibly helpful in the diagnostic process. We still do not know _HOW_ you are being redirected to a different endpoint than what I'm getting redirected to with SWAG the logs or configuration may have revealed this and other issues. It was also not an unreasonable request, and quite unreasonable to not provide them when requested. Just because we managed to find it in this instance after many hours does not negate these facts. It's obviously hard to say for certain since we don't have time machines, and we still do not know what they would have revealed if anything.

It is also not a reasonable excuse for your behavior. You're extremely close to burning all of your bridges with me. I would suggest taking a step back and consider how you've treated me.

**Comment by HannesJo0139** [comment]
I know there is a reason why opening issues without offering configs or logs is generally discouraged. Still I think it should be possible to read an issue and think about it before insisting on strict compliance even when it may be useless in a certain case. I provided a simple 2 steps list of how to reproduce. If you were unable to reproduce you simply did not do what is stated there. You can read step 2: Open my-site.org/authelia. So instead of asking for config and logs you could have asked why I want to open my-site.org/authelia. Or you could have handled it the way [nightah](https://github.com/nightah) did, who actually solved the whole issue withing 2 minutes.

You may have edited some of your comments to make it less obvious, but you seem to view the issues section as a service you provide to us while spending your freetime. This is not the case. On the contrary, users are spending their freetime to provide you with information that may help with your project. I have never been here in search for help. I'm here to help you fixing stuff. Closing issues because you insist on strict compliance with the rules even when unnecessary is not a problem for me, but it is for your project.

Stating that my post was no reasonable excuse even clarifies your position. I did not ask for an apology. I'm just treating you with the same attitude you suddenly started to show after not reading the issue properly. Again, I'm not the one who benefits from any bridges. You benefit from users reporting issues. But since it doesnt make me happy dealing with such an attitude I will simply switch to sth that does. Bye.
